From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Redned <redned235@gmail.com>
Date: Mon, 12 Jun 2023 11:54:12 -0500
Subject: [PATCH] Fix incorrect crafting result amount for fireworks

Although vanilla does not specifically call this method anywhere, this fixes a bug where the result using the Bukkit API returns the wrong amount.

diff --git a/src/main/java/net/minecraft/world/item/crafting/FireworkRocketRecipe.java b/src/main/java/net/minecraft/world/item/crafting/FireworkRocketRecipe.java
index 129f16dc1416732b7a2121346b0571f475c48c97..e5035cce1df5a10bc5a7216156823abe0454f257 100644
--- a/src/main/java/net/minecraft/world/item/crafting/FireworkRocketRecipe.java
+++ b/src/main/java/net/minecraft/world/item/crafting/FireworkRocketRecipe.java
@@ -81,7 +81,7 @@ public class FireworkRocketRecipe extends CustomRecipe {
 
     @Override
     public ItemStack getResultItem(RegistryAccess registryManager) {
-        return new ItemStack(Items.FIREWORK_ROCKET);
+        return new ItemStack(Items.FIREWORK_ROCKET, 3); // Paper - Fix incorrect crafting result amount
     }
 
     @Override
